cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -307,6 +309,65 @@ class SimpleArray
         }
     }
 
+    explicit SimpleArray(small_vector<size_t> const & shape,
+                         small_vector<size_t> const & stride,
+                         std::shared_ptr<buffer_type> const & buffer,
+                         bool is_c_contiguous = true,
```

Comment:
why not use an enum here? @ThreeMonth03

Reply:
I realized I could use an enum class inside the function after I asked the question.

Reply:
@ThreeMonth03  Maybe you can consider opening another PR to fix this.